cobalt/build/configs/BUILD.gn

```diff
@@ -24,22 +25,28 @@ config("cobalt_config") {
     configs += [ "//${starboard_path}/platform_configuration" ]
   } else if (is_cobalt_hermetic_build &&
              current_toolchain == cobalt_toolchain) {
-    configs += [ "//starboard/build/config:starboard" ]
-    if (sb_is_evergreen) {
-      if (current_cpu == "arm") {
-        # TODO: b/419893744 - Find a way to switch between hardfp, softfp.
-        configs += [ "//starboard/evergreen/arm/hardfp/platform_configuration" ]
-      } else if (current_cpu == "x64") {
-        configs += [ "//starboard/evergreen/x64/platform_configuration" ]
-      }
-    } else {
-      configs += [
-        # TODO: b/390500840 - Remove hardcoded x64 config, switch config based on platform.
-        "//starboard/build/config/modular/x64",
-      ]
+    configs += [
+      "//starboard/build/config:starboard",
+      ":modular_platform_configuration",
+    ]
+  }
+}
+
+config("modular_platform_configuration") {
+  cpu_path = current_cpu
+  if (current_cpu == "arm") {
+    cpu_path += "/$arm_float_abi"
+    if (arm_float_abi == "hard") {
+      cpu_path += "fp"
```

Comment:
I would prefer to use arm_float_abi as is and have our directories renamed to "hard" from "hardfp", but that's a much bigger change and I'm not sure it's worth it